### PR TITLE
fix: adding doc note about admin ingress

### LINKED ISSUE #33203: Explicitly document that the Operator does not create an Ingress for Admin URL
### Description

Add a note to the [Operator guide](https://github.com/keycloak/keycloak/blob/75eafead72275ff8fe1aacabf4469ba8b0b62df5/docs/guides/operator/basic-deployment.adoc?plain=1#L190) that the `.spec.hostname.admin` does not create additional Ingress and one needs to be created manually.

### Discussion

_No response_

### Motivation

It is confusing that `.spec.hostname.hostname` creates an Ingress and `.spec.hostname.admin` does not.

### Details

It does not make much sense for the Operator to create an Admin Route as it usually requires deeper configuration and that is specific to a given Ingress implementation which is out of scope for the Operator at the moment.

### PR DESCRIPTION
**Pull request by shawkins**
closes: #33203

### CODE REVIEW

File: docs/guides/operator/basic-deployment.adoc [hunk 1]
```diff
@@ -100,6 +100,10 @@ When running on OpenShift, with ingress enabled, and with the spec.ingress.class
 The operator will assign a default hostname to the stored version of the CR similar to what would be created by an OpenShift Route without an explicit host - that is ingress-namespace.appsDomain
 If the appsDomain changes, or should you need a different hostname for any reason, then update the Keycloak CR.
 
+NOTE: If you set the `hostname-admin`, or the deprecated `hostname-admin-url`, even if you enable ingress, no ingress will be created specifically for admin access.
```
Comment (vmuzikar) [inline docs/guides/operator/basic-deployment.adoc:103]: `hostname-admin-url` has been removed already.
```suggestion
NOTE: If you set the `hostname-admin` even if you enable ingress, no ingress will be created specifically for admin access. 
```

**Review by vmuzikar — CHANGES_REQUESTED** [review]

File: docs/guides/operator/basic-deployment.adoc (hunk 1 above)
Comment (shawkins) [reply docs/guides/operator/basic-deployment.adoc:103]: > `hostname-admin-url` has been removed already.

Not from the operator #32485

Or do you want to ignore this case for the purposes of this note?

Comment (vmuzikar) [reply docs/guides/operator/basic-deployment.adoc:103]: Good point. Ok let's leave as is.

**Review by vmuzikar — DISMISSED** [review]

File: docs/guides/operator/basic-deployment.adoc [hunk 2]
```diff
@@ -100,6 +100,10 @@ When running on OpenShift, with ingress enabled, and with the spec.ingress.class
 The operator will assign a default hostname to the stored version of the CR similar to what would be created by an OpenShift Route without an explicit host - that is ingress-namespace.appsDomain
 If the appsDomain changes, or should you need a different hostname for any reason, then update the Keycloak CR.
 
+NOTE: If you set the `hostname-admin`, or the deprecated `hostname-admin-url`, even if you enable ingress, no ingress will be created specifically for admin access. 
+Admin access via a separate hostname is generally expected to have access restrictions, which are not currently expressible via the Keycloak CR. 
+Also the default ingress does not prevent accessing admin endpoints, so you may not want to enable ingress handling via the Keycloak CR at all when you have a separate hosname for admin endpoints.
```
Comment (mabartos) [inline docs/guides/operator/basic-deployment.adoc:105]:
```suggestion
Also the default ingress does not prevent accessing admin endpoints, so you may not want to enable ingress handling via the Keycloak CR at all when you have a separate hostname for admin endpoints.       
```

Small typo

**Review by vmuzikar — APPROVED** [review]

**Review by mabartos — APPROVED** [review]
@shawkins Thanks!

**Pull request merged**